### PR TITLE
feat: styling tabs + counter

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Adds inline search in the hooks list
- Styles the different screens, in particular when adding a custom hook
- Create placeholder logic and styles for simulation success/fail
- Two tabs, showing all hooks vs. the user's custom hooks + option to add their custom hook
- Tabs have a hook counter in the title
- If a hook has a success/fail simulation, it can handle optional passed Tenderly simulation URLs

Screenshot 2024-09-23 at 11 46 49
Screenshot 2024-09-23 at 11 46 43
Screenshot 2024-09-23 at 11 46 26
Screenshot 2024-09-23 at 11 46 20
Screenshot 2024-09-23 at 11 46 13
Screenshot 2024-09-23 at 11 45 51
Screenshot 2024-09-23 at 11 45 41
Screenshot 2024-09-23 at 11 45 30
Screenshot 2024-09-23 at 11 45 23
Screenshot 2024-09-23 at 11 45 14

Screenshot 2024-09-23 at 13 06 03
Screenshot 2024-09-23 at 12 49 57
Screenshot 2024-09-23 at 12 49 45
Screenshot 2024-09-23 at 12 49 29
Screenshot 2024-09-23 at 12 48 29
Screenshot 2024-09-23 at 12 47 05

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/hooksStore/containers/HookRegistryList/index.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ export function HookRegistryList({ onDismiss, isPreHook, hookToEdit }: HookStore
     if (!hookToEditDetails) {
       setSelectedDapp(null)
     } else {
-      setSelectedDapp(findHookDappById(dapps, hookToEditDetails) || null)
+      setSelectedDapp(findHookDappById(dapps, { ...hookToEditDetails, dappId: hookToEditDetails.dappId }) || null)
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/hooksStore/containers/HookRegistryList/index.tsx:77]: Seems the same as before :) Do I miss something?

File: apps/cowswap-frontend/src/modules/hooksStore/containers/HookRegistryList/index.tsx [hunk 2]
```diff
@@ -83,6 +83,9 @@ export function HookRegistryList({ onDismiss, isPreHook, hookToEdit }: HookStore
     setDappDetails(null)
   }, [isVerifiedHooksTab])
 
+  const verifiedHooksCount = isPreHook ? PRE_HOOK_REGISTRY[chainId].length : POST_HOOK_REGISTRY[chainId].length
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/hooksStore/containers/HookRegistryList/index.tsx:86]: 👍

**Comment by elena-zh** [comment]
Hey @fairlighteth , the build has failed. Could you please check it?

**Comment by fairlighteth** [comment]
@elena-zh Can you check again?

**Comment by elena-zh** [comment]
Hey @fairlighteth , great, great job!

However, I've noticed some issues: 
1. (reported in https://github.com/cowprotocol/cowswap/pull/4910#issuecomment-2368044964, case 3): : it would be great to truncate preceding and trailing spaces when adding a hook URL
image
2. When I add a custom hook into a post-hooks section, it apepars in the pre-hook section as well --> so I can add it there. But according to the manifest this should not be possible
image
image
3. When I edit an added hook, I can see a hook store with 2 tabs. I'd expect here to see an added hook details only for editing it
image
4. There are excessive spaces in the validation message: 
excessive spaces
5. A scrollbar appears at the bottom of a hook details in a mobile view. Is this an expected behavior? 
mobile view
6. (might be not an issue): Is this OK that 'details' start with the lower0case letter?
image
7. Gas limit validation: I can input 0, and the app accepts it. Might it be possible to add a validation for this case? 
0- succesful
8. I have not been able to get a failed simulation report even when I entered an invalid data. Could you please clarify how I can simulate this? 
wrong hooks fail
9. I think that it would be great to show a simulation report for a success outcome.
10. When I place an order with a custom recipient, I see this in the Account modal. 
account modal
11. Might it be possible to fetch a wallet type for a custom hook from its Manifest? 
image
image


Thanks!

**Comment by shoom3301** [comment]
@elena-zh thanks!

1. Fixed
2. Fixed
3. Fixed
4. Fixed
5. It's ok
6. It's ok
7. It's ok
8. This was a just layout without logic, I reverted this, we will add it back once the logic is ready
9. See 8
10. It's ok for now, we will improve it later

**Review by elena-zh — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @shoom3301 , thank you for the replies and fixes!

Some issues that I've faced: 
1. Impossible to fill in a custom recipient field on any form but Hooks. See the video: https://www.loom.com/share/e9123bab4d3943d3bb2671c04e987e73
2. When I add a custom hook, it should be displayed in the All tab (see this [discussion](https://cowservices.slack.com/archives/C0361CDG8GP/p1726839015909459))
image
3. Hooks should be hidden for ETH-flow orders. When a hook has been added, and a trade is changed to an ETH-order, I can see the recipient field still prefilled. It would be great to clear out this field to prevent accidental funds losses. 
image
4. Edit hook: 
 - custom: does not open an added hook details, just shows a Hook store
edit custom
- Claim GNO: when edit it, it adds a new hook to the form when save changes

Could you please take a look at these issues?

**Comment by shoom3301** [comment]
@elena-zh 

1. Fixed
2. Fixed
3. Fixed
4. Sorry, I don't understand the problem, could you give more information?
5. Can't reproduce
6. Can't reproduce

**Comment by shoom3301** [comment]
@elena-zh 5th is also fixed

**Pull request merged**